tests/suites/test_suite_platform.function

```diff
@@ -84,7 +89,22 @@ void time_delay_seconds(int delay_secs)
     sleep_ms(delay_secs * 1000);
 
     elapsed_secs = mbedtls_time(NULL) - current;
-    TEST_ASSERT(elapsed_secs >= delay_secs && elapsed_secs < 4 + delay_secs);
+
+    /*
+     * `mbedtls_time` was defined as c99 function `time`, it uses
+     * CLOCK_REALTIME and returns the number of seconds since the Epoch. And
+     * `nanosleep` uses CLOCK_MONOTONIC. The time sources are out of sync.
+     *
+     * If CLOCK_MONOTONIC is faster than CLOCK_REALTIME and `nanosleep` exits at
```

Comment:
Surely it's not really about whether one clock is "faster" than the other (I would expect them to tick at the same rate all other things being equal) but if one is adjusted and the other not - which is what happens when one is a "wallclock" timer and the other is a monotonic timer

Reply:
Yes, that's more easily understand. I will change that.

But I think this problem can be abstracted into two different rate clock problems. The wall clock is come from remote and another one come from local. Due to implementation issues，wall clock shows discontinue jumps problem. If wall clock is updated very frequently，discontinue jumps will disappear.  And user will get two different rate clocks.

Reply:
> The wall clock is come from remote

I don't understand. "Wall clock" in this context means that this particular timer from the kernel should match as closely as possible to the time that someone looking at their watch would see. So time may be stepped forwards or backwards as daylight savings happens (for example). A monotonic clock must by definition only ever increase.

The ticks that advance these clocks come from local hardware, which may not be precise. So there is frequency adjustment, which should affect all clocks, so that when each clock says "one second has passed" as close to one second has possible has actually passed

Reply:
>"Wall clock" in this context means that this particular timer from the kernel should match as closely as possible to the time that someone looking at their watch would see. 

I mean it appears to be from a remote server if updated fast enough. 

> So time may be stepped forwards or backwards as daylight savings happens (for example). A monotonic clock must by definition only ever increase.

I do not think  `daylight savings` will affect the value of `time()` , for it is `the number of seconds since the Epoch` . 

> which should affect all clocks

No. `CLOCK_BOOTTIME` and `CLOCK_*_CPUTIME_ID` will not be affected by time adjustment. `CLOCK_MONOTONIC is affected by the incremental adjustments performed`, that's different with CLOCK_REALTIME. 
If decreasing adjustment peformed, CLOCK_REALTIME will change and CLOCK_MONOTONIC will not change.